src/config_store.rs

```diff
@@ -82,26 +88,50 @@ impl ConfigStore {
     pub fn add_config(&mut self, config: Arc<CheckConfig>) {
         self.configs.insert(config.subscription_id, config.clone());
 
+        let bucket = self
+            .partitioned_buckets
+            .entry(config.partition)
+            .or_insert_with(|| (0..MAX_CHECK_INTERVAL_SECS).map(|_| vec![]).collect());
+
         // Insert the configuration into the appropriate slots
         for slot in config.slots() {
-            self.buckets[slot].insert(config.subscription_id);
+            bucket[slot].push(config.clone());
         }
     }
 
     /// Remove a Check Configuration from the store.
     pub fn remove_config(&mut self, subscription_id: Uuid) {
-        if let Some(config) = self.configs.remove(&subscription_id) {
-            for slot in config.slots() {
-                self.buckets[slot].remove(&subscription_id);
-            }
+        let Some(config) = self.configs.remove(&subscription_id) else {
+            return;
+        };
+        let Some(buckets) = self.partitioned_buckets.get_mut(&config.partition) else {
+            return;
+        };
+
+        for slot in config.slots() {
+            buckets[slot].retain(|c| c.subscription_id != subscription_id);
```

Comment:
Isn't using retain here inefficient? Should we have a data structure that allows o(1) lookups?